pkg/cloud/services/secretsmanager/secret.go

```diff
@@ -50,40 +58,57 @@ func (s *Service) Create(m *scope.MachineScope, data []byte) (string, error) {
 		Additional:  additionalTags,
 	})
 
-	name := s.secretName(m)
+	chunks := 1
+	buf := bytes.NewBuffer(data)
+	prefix := s.secretNamePrefix(m)
 
-	resp, err := s.scope.SecretsManager.CreateSecret(&secretsmanager.CreateSecretInput{
-		Name:         aws.String(name),
-		SecretBinary: data,
-		Tags:         converters.MapToSecretsManagerTags(tags),
-	})
+	for {
+		name := fmt.Sprintf("%s-%d", prefix, chunks-1)
+
+		chunk := buf.Next(maxSecretSizeBytes)
+
+		_, err := s.scope.SecretsManager.CreateSecret(&secretsmanager.CreateSecretInput{
+			Name:         aws.String(name),
+			SecretBinary: chunk,
+			Tags:         converters.MapToSecretsManagerTags(tags),
+		})
+
+		if err != nil {
+			return "", 0, err
+		}
 
-	if err != nil {
-		return "", err
+		if len(chunk) < maxSecretSizeBytes {
+			break
+		}
+
+		chunks++
```

Comment:
Do we need to add some type of attempt to cleanup in the event that we get a failure part way through the creation of the secrets?

Reply:
Added some mop up in the controller part